### ISSUE TITLE
Running Tasks Are Using Incomplete Export Path

### ISSUE THREAD

**Issue by irwansyahwii** [Microsoft/vscode#15283]
- VSCode Version: Code 1.7.1 (02611b40b24c9df2726ad8b33f5ef5f67ac30b44, 2016-11-03T13:46:46.244Z)
- OS Version: Linux x64 4.4.0-43-generic
- Extensions:

|Extension|Author|Version|
|---|---|---|
|output-colorizer|IBM|0.0.8|
|githistory|donjayamanne|0.1.0|
|docthis|joelday|0.3.5|
|debugger-for-chrome|msjsdiag|2.2.1|
|gitflow|vector-of-bool|0.1.0|

---

I have a commander.sh file which will run another .sh file based on the parameter given. It was running fine until yesterday. When I ran it, it said:

running command$ ./commander.sh compile-backend
TASK BEGIN...
Thu Nov 10 14:03:02 WIB 2016
compile-backend
TASK FINISHED!
./install-model.sh: line 1: npm: command not found
./install-service.sh: line 1: npm: command not found
./commander.sh: line 43: npm: command not found
Thu Nov 10 14:03:02 WIB 2016

When I printed the export PATH the running task was not have a configured PATH variable that's why it can't found npm and other cli commands. I am using Linux Mint and I had run the updater prior this error.  So probably because of linux's system update but I still don't know how to fix this problem. Can anyone help? The only way that I know is to clone the VSCode source code and try to debug the problem myself but the npm install on the repo is very big and took a long time and still not finished by the time I am writing this issue

**Comment by irwansyahwii** [comment]
One more thing. When I ran the .sh file directly using Linux terminal it was running fine. So the problem only happening when I ran it from VS Code tasks management (tasks.json).

**Comment by dbaeumer** [comment]
@irwansyahwii how do you start VS Code? By clicking a icon on a desktop or from a terminal. If the first can you please check if the same reproduces when starting from a terminal?

And can you please check the following as well: start VS Code and do Help  > Toggle Developer Tools. In the developer tools console type `process.env.Path`. Does the value contain the path to the `npm` command.

**Comment by dbaeumer** [comment]
@joaomoreno FYI.

**Comment by irwansyahwii** [comment]
Hi @dbaeumer  I always ran it using the application icon from the menu. I just tried running VSCode from terminal and the task can execute npm but the tsc can't recognize all the npm modules :(

**Comment by dbaeumer** [comment]
What do you mean by:

>  tsc can't recognize all the npm modules :( 

And can you please do

> And can you please check the following as well: start VS Code and do Help > Toggle Developer Tools. In the developer tools console type process.env.Path. Does the value contain the path to the npm command.

**Comment by joaomoreno** [comment]
@irwansyahwii What do you mean by _It was running fine until yesterday_? What happened _yesterday_? Did you update Code from an old version to the current 1.7.1?

**Comment by irwansyahwii** [comment]
@dbaeumer  my .sh file calling tsc to compile all the source code in the current folder. Before that it will do npm install for a specific package. When I ran the VS Code from menu the task will says it can't find npm and tsc command. But when I ran from a terminal it can execute npm successfully but the tsc (typescript compiler) said it can't find all the npm modules reference by my codes. I did what you asked to print the environtment path from the devtools console and the result is different compare to running from terminal. 

This is the result from running from the menu:
"/home/irwansyah/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"

and this is from running from a terminal:
"/home/irwansyah/.nvm/versions/node/v6.3.1/bin:/home/irwansyah/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/irwansyah/Android/Sdk/tools:/usr/lib/jvm/java-8-oracle/"

The one from the menu doesn't contains any path pointing to node folder so that's why it can't run npm but the one from terminal is. I don't mind to run it from terminal but still the tsc can't compile properly.

@joaomoreno I believe the previous day I execute Linux Mint system update. After that I have to restart VSCode since the intelllisense will took a long time after I am working with many files so I have restart it so that it fast again. But after restarting suddenly the tasks missing many system paths. I tried to reinstall, then downgrade to 1.6, then reinstall 1.7 again but still all the versions can't execute the task command properly.
